### PR TITLE
Fixup improperly scoped variables in ImbuModels class

### PR DESCRIPTION
**Pull request by oxtopus**
Required for pending changes to imbu fluent_api.py in numenta-apps

### CODE REVIEW

File: htmresearch/frameworks/nlp/imbu.py [hunk 1]
```diff
@@ -127,20 +133,25 @@ def _modelFactory(self, modelType, **kwargs):
     kwargs.update(**self._defaultModelFactoryKwargs())
 
     if modelType == "CioWordFingerprint":
-      kwargs.update(retina=kwargs.get("retina") or self.defaultRetina,
-                    apiKey=kwargs.get("apiKey"),
+      kwargs.update(modelDir=savePath,
+                    retina=self.retina,
+                    apiKey=self.apiKey,
```
Comment (BoltzmannBrain) [inline htmresearch/frameworks/nlp/imbu.py]: If `modelDir`, `retina`, and `apiKey` are in all the if blocks, couldn't they just be in one `kwargs` update or included in `self._defaultModelFactoryKwargs()`?

File: htmresearch/frameworks/nlp/imbu.py [hunk 2]
```diff
@@ -89,22 +89,29 @@ class ImbuModels(object):
   defaultModelType = "CioWordFingerprint"
   defaultRetina = "en_associative"
 
-  def __init__(self, cacheRoot, dataPath, loadPath, savePath,
-      modelSimilarityMetric):
+  # Sequence of classification model types that accept CioEncoder kwargs
+  requiresCIOKwargs = {
```
Comment (BoltzmannBrain) [inline htmresearch/frameworks/nlp/imbu.py]: Where is this used?

**Comment by BoltzmannBrain** [comment]
@oxtopus done with review, but haven't yet looked at the numenta-apps PR.

File: htmresearch/frameworks/nlp/imbu.py (hunk 1 above)
Comment (oxtopus) [inline htmresearch/frameworks/nlp/imbu.py]: It's only considered for types that interact with cortical.io, I.e. these three. Iirc, not all types do, and there may be more in the future.  If we make it required for all models, then yes, we can move it back into `_defaultModelFactoryKwargs()` as it was before, but I don't think that's the right pattern moving forward.

File: htmresearch/frameworks/nlp/imbu.py (hunk 2 above)
Comment (oxtopus) [inline htmresearch/frameworks/nlp/imbu.py]: In fluent_all.py to determine whether or not to pass cortical.io credentials, but in retrospect, could be used in _model factory() here, too.  Will follow up tomorrow when I have my computer.

**Pull request merged**